src/Chip/ChipIcon.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { KeyboardEventHandler, MouseEventHandler } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../Icon';
+// @ts-ignore
+import IconButton from '../IconButton';
+// @ts-ignore
+import { STYLE_VARIANTS } from './constants';
+
+export interface ChipIconProps {
+  className: string,
+  src: React.ReactElement | Function,
+  onClick?: KeyboardEventHandler & MouseEventHandler,
+  alt?: string,
+  variant: string,
+}
+
+function ChipIcon({
+  className, src, onClick, alt, variant,
+}: ChipIconProps) {
+  if (onClick) {
+    return (
+      <IconButton
+        className={className}
+        src={src}
+        onClick={onClick}
+        onKeyPress={onClick}
+        iconAs={Icon}
+        alt={alt}
+        invertColors={variant === STYLE_VARIANTS[1]}
+      />
+    );
+  }
+
+  return <Icon src={src} className={className} />;
+}
+
+ChipIcon.propTypes = {
+  className: PropTypes.string.isRequired,
+  src: PropTypes.oneOfType([PropTypes.element, PropTypes.func]).isRequired,
```

Comment:
[curious] Could this prop type be reduced to `PropTypes.elementType`?

> // A React element type (eg. MyComponent).
// a function, string, or "element-like" object (eg. React.Fragment, Suspense, etc.)
// see https://github.com/facebook/react/blob/master/packages/shared/isValidElementType.js

Reply:
I don't think so, `elementType` will also allow you to pass strings like `'div'` which we do not want here. We usually use `elementType` type for  the `as` prop that allows you to specify base element for the component, but here I believe we only want actual React components, e.g. our icons.